signature/cose/envelope.go

```diff
@@ -328,78 +311,47 @@ func getSignatureAlgorithmFromKeySpec(keySpec signature.KeySpec) (cose.Algorithm
 	}
 }
 
-func generateCoseProtectedHeaders(req *signature.SignRequest, protected cose.ProtectedHeader) error {
-	// crit, expiry, signingTime
-	var crit []interface{}
+func generateProtectedHeaders(req *signature.SignRequest, protected cose.ProtectedHeader) error {
+	// crit, signingScheme, expiry, signingTime, authenticSigningTime
+	crit := []interface{}{headerLabelSigningScheme}
+	protected[headerLabelSigningScheme] = string(req.SigningScheme)
+	switch req.SigningScheme {
```

Comment:
not a strong opinion, I feel we can have a map mapping from the schema to the header label to avoid the switch. Line 378 can benefit from this map as well.